### PR TITLE
[#17] Test cleanup of Rotator.

### PR DESCRIPTION
**Pull request by 365andreas**

### CODE REVIEW

File: test/Cardano/BM/Test/Rotator.lhs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    n = 5
+    rot num = RotationParameters
+              { rpLogLimitBytes = 10000000 -- 10 MB
+              , rpMaxAgeHours   = 24
+              , rpKeepFilesNum  = num
+              }
+
+\end{code}
+
+\subsubsection{Check that name giver only adds 15 digits to the name of the file.}\label{code:prop_name_giving}
```
Comment (CodiePP) [inline test/Cardano/BM/Test/Rotator.lhs]: Check that full file name has only added 15 digits to the base name of the file.

File: test/Cardano/BM/Test/Rotator.lhs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+\begin{code}
+prop_name_giving :: FilePath -> Property
+prop_name_giving name = monadicIO $ do
+    filename <- run $ nameLogFile name
+    assert $ length filename == length name + 15
+
```
Comment (CodiePP) [inline test/Cardano/BM/Test/Rotator.lhs]: have you printed out some of these generated filenames? just curious.

File: test/Cardano/BM/Test/Rotator.lhs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+\begin{code}
+
+data LocalFilePath = Dir FilePath
+    deriving (Show)
+
+instance Arbitrary LocalFilePath where
+    arbitrary = do
+        start <- QC.sized $ \n -> replicateM n (QC.elements $ ['a'..'z'])
+        x <- QC.sized $ \n -> replicateM n (QC.elements $ ['a'..'z'] ++ "/")
+        pure $ Dir ("./test-rot" </> (start ++ x))
```
Comment (CodiePP) [inline test/Cardano/BM/Test/Rotator.lhs]: maybe put  all this in a directory returned by `getTemporaryDirectory`

File: test/Cardano/BM/Test/Rotator.lhs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+\begin{code}
+prop_name_giving :: FilePath -> Property
+prop_name_giving name = monadicIO $ do
+    filename <- run $ nameLogFile name
+    assert $ length filename == length name + 15
+
+\end{code}
+
+\subsubsection{Check cleanup of rotator.}\label{code:prop_cleanup}
+\begin{code}
```
Comment (CodiePP) [inline test/Cardano/BM/Test/Rotator.lhs]: can you describe in details what this test wants to achieve?
clean up in the context of log rotator could also mean that it removes old log files (by time, or the number of files to keep).
The test `prop_cleanup` will sample over (RotationParameters x LocalFilePath x Small Int) which probably is quite huge!
Could the test be more specific?

File: test/Cardano/BM/Test/Rotator.lhs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    arbitrary = do
+        start <- QC.sized $ \n -> replicateM n (QC.elements $ ['a'..'z'])
+        x <- QC.sized $ \n -> replicateM n (QC.elements $ ['a'..'z'] ++ "/")
+        pure $ Dir ("./test-rot" </> (start ++ x))
+
+    shrink _ = []
+
+prop_cleanup :: RotationParameters -> LocalFilePath -> Small Int -> Property
+prop_cleanup rotationParams (Dir path) (Small n) = monadicIO $ do
+    dates <- run $ replicateM (abs n) $ QC.generate $ replicateM 14 (QC.elements ['0'..'9'])
```
Comment (CodiePP) [inline test/Cardano/BM/Test/Rotator.lhs]: this does not lead to dates which the log rotator could interpret.

File: test/Cardano/BM/Test/Rotator.lhs (hunk 4 above)
Comment (365andreas) [reply test/Cardano/BM/Test/Rotator.lhs]: `RotationParameters` is fixed to:
```
RotationParameters
              { rpLogLimitBytes = 10000000 -- 10 MB
              , rpMaxAgeHours   = 24
              , rpKeepFilesNum  = num
              }
```

**Comment by 365andreas** [comment]
WIP enhance generator:
1. checking exactly which files should remain
2. generetang long time intervals to check that files are also removed when they have aged too much

**Pull request merged**